kobo/xmlrpc.py

```diff
@@ -481,7 +481,8 @@ def request(self, *args, **kwargs):
                 except (socket.error, socket.herror, socket.gaierror, socket.timeout,
                         httplib.CannotSendRequest, xmlrpclib.ProtocolError) as ex:
                     if i >= self.retry_count:
-                        raise
+                        print(f"XML-RPC connection to {args[0]} failed: {ex}", file=sys.stderr)
+                        sys.exit(1)
```

Comment:
I would not expect `retry_request_decorator()` to terminate the process when a connection fails.  That feels totally wrong.  If such behavior is desired, the re-raised exception needs to be caught again at a higher level and handled as needed.

Reply:
exceptions occur when initializing a hub proxy. I'm doubting whether it's worth a fix at all since it needs to quit somewhere even at a higher level.

I reverted this and moved the exception handling upper.